src/rbr/RBR.java

```diff
@@ -129,24 +129,28 @@ public void regionsComputation() {
         assert reachabilityIsOk();
     }
 
-    private void adjustRegions(Vertex sw) {
+    void adjustRegions(Vertex sw) {
         List<Region> newRegions = new ArrayList<>();
-        List<Region> regionsToBeRemoved = new ArrayList<>();
         for (Region currentRegion : regionsForVertex.get(sw)) {
-            Set<Vertex> outsiders = currentRegion.outsiders();
-            if(outsiders.isEmpty())
-                continue;
+            newRegions.addAll(adjustedRegionsFrom(currentRegion));
+        }
+        regionsForVertex.put(sw, newRegions);
+    }
+
+    ArrayList<Region> adjustedRegionsFrom(Region r) {
+        ArrayList<Region> adjustedRegions = new ArrayList<>();
+        Set<Vertex> outsiders = r.outsiders();
+        if(outsiders.isEmpty()){
+            adjustedRegions.add(r);
+        } else {
             Range outsidersBox = TopologyKnowledge.box(outsiders);
-            Set<Vertex> trulyDestinationsInOutsidersRange = currentRegion.destinationsIn(outsidersBox);
+            Set<Vertex> trulyDestinationsInOutsidersRange = r.destinationsIn(outsidersBox);
 
-            regionsToBeRemoved.add(currentRegion);
-            List<Region> regionsToAdd = splitRegionExcludingOutsiders(currentRegion, outsidersBox);
-            newRegions.addAll(regionsToAdd);
-            // use others routers to make others regions
-            newRegions.addAll(makeRegions(trulyDestinationsInOutsidersRange, currentRegion.inputPorts(), currentRegion.outputPorts()));
+            List<Region> regionsToAdd = splitRegionExcludingOutsiders(r, outsidersBox);
+            adjustedRegions.addAll(regionsToAdd);
+            adjustedRegions.addAll(makeRegions(trulyDestinationsInOutsidersRange, r.inputPorts(), r.outputPorts()));
```

Comment:
I actually would prefer no if/else at all here. Although it can be a next step (as would require 'fix' 'splitRegionExcludingOutsiders' and 'makeRegions') is a good practice to minimize things inside conditionals, so go ahead.